packages/framework/fluid-static/src/fluidContainer.ts

```diff
@@ -245,7 +259,21 @@ export function createFluidContainer<
 }
 
 /**
- * Base {@link IFluidContainer} implementation.
+ * Check that the provided `container` is an internal {@link IFluidContainerInternal}.
+ *
+ * @internal
+ */
+export function isInternalFluidContainer<TContainerSchema extends ContainerSchema>(
```

Comment:
I wonder if it might be better (or at least easier) to just export the `FluidContainer` class as `@internal`, and make the `container` property public.

This pattern also seems fine, but also seems unnecessary since this is for internal use only.

Reply:
In general, I think we should hide classes, even internal ones. `/internal` doesn't mean inaccessible even if no one outside should. When there isn't a class exposed, things are just that much safer/stable. Easier to expose it eventually if we need that than to walk-it back.
(I see Bohemia using /internal things even though they shouldn't.)

Reply:
Yeah, that's fine. And we can easily change this in the future if we want to.